mlir/include/mlir/Dialect/MemRef/IR/MemRefOps.td

```diff
@@ -1351,7 +1351,8 @@ def MemRef_PrefetchOp : MemRef_Op<"prefetch"> {
     instruction cache.
   }];
 
-  let arguments = (ins AnyMemRef:$memref, Variadic<Index>:$indices,
+  let arguments = (ins Arg<AnyMemRef, "prefetch address", [MemWrite]> :$memref,
```

Comment:
Why do you need to "write to the buffer" semantically here?
(it's worth a comment, as this may not be intuitive!)

Reply:
@ftynse I believe the source of the problem is that without side-effects defined, the buffer allocation will fail:

> "ops with unknown memory side effects are not supported"

So we need to add side effects, and the part about 
> It's basically to ensure that dead-code analysis does not remove the operation

is only an answer to "why write and not just read".

Reply:
An alternative design here, more SSA-friendly, would be for the op to be side-effect free and return a token to be consumed later by some load operation.